### PR TITLE
update eval.lua and bake doc gen into cmake

### PR DESCRIPTION
**Pull request by lewis6991**
- fix(types): add more annotations to eval.lua
- docs(generators): bake into cmake

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -173,4 +173,9 @@ $(DEPS_BUILD_DIR)/%: phony_force
 	$(BUILD_TOOL) -C $(DEPS_BUILD_DIR) $(patsubst $(DEPS_BUILD_DIR)/%,%,$@)
 endif
 
+.PHONY: doc
+doc: api-metadata
```
Comment (dundargoc) [inline Makefile:177]: Specify dependencies with cmake instead of make. Use make only for the top level command as syntactic sugar.

**Review by dundargoc — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (lewis6991) [reply Makefile:177]: help?

Comment (lewis6991) [reply Makefile]: Ok, I think I've figured it out and made it good.

**Review by dundargoc — APPROVED** [review]
Looks good. I haven't been able to test the code but I've read it and believe it should work.

**Pull request merged**